### PR TITLE
Alert System and FlashMaps BNN-180 #resolve

### PR DESCRIPTION
**Pull request by nurv**
May need further enhancements, including moving classes to appropriate packages, but it provides the initial foundations so we can start getting the conversation rolling.

### CODE REVIEW

**Comment by jcarvalho** [comment]
I think that the alerts deserve their own package, and the internal classes (like FlashMap) should be moved there as package-private.

**Comment by nurv** [comment]
@jcarvalho I'm following the [nomenclature](https://confluence.fenixedu.org/display/FENIXEDU/Methodology) that we've agreed upon. Should I create the `org.fenixedu.bennu.alerts` package?

File: bennu-portal/src/main/java/org/fenixedu/bennu/portal/servlet/PortalLayoutInjector.java [hunk 1]
```diff
@@ -89,6 +91,7 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain
         PortalResponseWrapper wrapper = new PortalResponseWrapper(response);
 
         chain.doFilter(request, wrapper);
+        JsonArray alerts = Alert.getAlertsAsJson(request, response);
```
Comment (jcarvalho) [inline bennu-portal/src/main/java/org/fenixedu/bennu/portal/servlet/PortalLayoutInjector.java]: Please inline this variable.

**Comment by jcarvalho** [comment]
Only have a single pending comment, otherwise :+1:

**Comment by nurv** [comment]
That comment has been resolved, can integrate.

**Pull request merged**